### PR TITLE
Polish OSDK documentation

### PR DESCRIPTION
**Pull request by StevenJiang1110**

### CODE REVIEW

**Comment by tatetian** [comment]
@junyang-zh Could you review this PR? Your feedbacks would be highly appreciated. Thanks!

File: OSDK.toml [hunk 1]
```diff
@@ -1,6 +1,14 @@
+# The common options for build, run and test
```
Comment (junyang-zh) [inline OSDK.toml:1]: Maybe add a top comment that says:

```toml
# The OSDK manifest at the Asterinas root virtual workspace to provide default OSDK settings for all packages.
```

File: docs/src/osdk/reference/commands/build.md [hunk 2]
```diff
@@ -29,6 +29,11 @@ Build artifacts in release mode, with optimizations
 - `--features <FEATURES>`:
 Space or comma separated list of features to activate
 
+- `--no-default-features`
+Do not activate the `default` features
+
+More Cargo options will be supported in future versions of OSDK.
```
Comment (junyang-zh) [inline docs/src/osdk/reference/commands/build.md:35]: The `--config` option of Cargo is supported as well. Maybe document it here too.

https://github.com/asterinas/asterinas/blob/e0c6c294815d23800f3437bbb1b755f635dec338/osdk/src/cli.rs#L238-L244

File: docs/src/osdk/reference/manifest.md [hunk 3]
```diff
@@ -35,61 +35,139 @@ path fields will be relative to the manifest's enclosing directory.
...
+[schema."custom".run]                       # <19>
+[schema."custom".test]                      # <20>
 ```
 
 Here are some additional notes for the fields:
 
-1. The arguments provided will be passed to the guest kernel.
+1. The type of current crate.
+
+    Optional. The default value is `kernel`.
```
Comment (junyang-zh) [inline docs/src/osdk/reference/manifest.md:88]: As the implementation now tells, the default value is inferred from the usage of `#[aster_main]` if not specified. The default value of `cargo new` is indeed `kernel`. Maybe document it.

File: docs/src/osdk/reference/manifest.md [hunk 4]
```diff
@@ -35,61 +35,139 @@ path fields will be relative to the manifest's enclosing directory.
...
-1. The arguments provided will be passed to the guest kernel.
+1. The type of current crate.
+
+    Optional. The default value is `kernel`. 
+    
+    Possible values are `library` or `kernel`.
+
+2. The architectures that can be supported.
+
+    Optional. By default, the architecture of the host machine will be used.
```
Comment (junyang-zh) [inline docs/src/osdk/reference/manifest.md:94]: By default OSDK will regard all the settings in this scheme (or the default scheme) to be architectural-independent. That is, supporting all architectures.

When building or running, if not specified in the CLI, the architecture of the host machine will be used.

**Review by junyang-zh — COMMENTED** [review]
Good job on improving the documentation! Here's minor suggestions.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**